src/Helpers/Components.php

```diff
@@ -1043,4 +890,483 @@ function ($a) use (&$return) {
+	 */
+	private static function getDefaultBreakpoints(array $breakpoints): array
+	{
+		return [
+			'min' => array_keys($breakpoints)[0] ?? '',
+			'max' => array_keys($breakpoints)[count($breakpoints) - 1] ?? '',
+		];
+	}
+
+	/**
+	 * Iterating through variables matching the keys from responsiveAttributes and translating it to responsive attributes names.
+	 *
+	 * @param array<string, mixed> $attributes Attributes that are read from component's/block's manifest.
+	 * @param array<string, array<string, mixed>> $variables Variables that are read from component's/block's manifest.
+	 * @param array<int|string, mixed> $data Predefined structure for adding styles to a specific breakpoint value.
+	 * @param array<string, mixed> $manifest Component/block manifest data.
+	 * @param array<string, mixed> $defaultBreakpoints Default breakpoints for mobile/desktop first.
+	 *
+	 * @return array<int|string, mixed> Array prepared for setting all the variables to its breakpoints.
+	 */
+	private static function setVariablesToBreakpoints(array $attributes, array $variables, array $data, array $manifest, array $defaultBreakpoints): array
+	{
+		foreach ($variables as $variableName => $variableValue) {
+			// Constant for attributes set value (in db or default).
+			$attributeValue = $attributes[self::getAttrKey($variableName, $attributes, $manifest)] ?? '';
+
+			// Make sure this works correctly for attributes which are toggles (booleans).
+			if (is_bool($attributeValue)) {
+				$attributeValue = $attributeValue ? 'true' : 'false';
+			}
+
+			// If type default or value.
+			if (!self::arrayIsList($variableValue)) {
+				$variableValue = $variableValue[$attributeValue] ?? [];
+			}
+
+			// Bailout if wrong type is provided.
+			if (!is_array($variableValue)) {
+				continue;
+			}
+
+			// Iterate variable array to check breakpoints.
+			foreach ($variableValue as $breakpointItem) {
+				// Define variables from breakpointItem.
+				$variable = $breakpointItem['variable'] ?? [];
+
+				// Check if we are using mobile or desktop first. Mobile first is the default.
+				$isInverse = $breakpointItem['inverse'] ?? false; // If inverse is not set use mobile first.
+				$type = $isInverse ? 'max' : 'min';
+
+				// If breakpoint is not set or if breakpoint is a default breakpoint use default name.
+				$isDefaultBreakpoint = empty($breakpointItem['breakpoint']) || $breakpointItem['breakpoint'] === $defaultBreakpoints[$type];
+				$breakpoint = $isDefaultBreakpoint ? 'default' : $breakpointItem['breakpoint'];
+
+
+				// Iterate each data array to find the correct breakpoint.
+				foreach ($data as $index => $item) {
+					// Check if breakpoint and type match.
+					if ($item['name'] === $breakpoint && $item['type'] === $type && (!empty((string) $attributeValue) || gettype($attributeValue) === 'integer' || gettype($attributeValue) === 'float' || gettype($attributeValue) === 'double' || $attributeValue === '0')) { // phpcs:ignore Generic.Files.LineLength.TooLong
```

Comment:
Maybe break these compound conditionals into newlines for readability, e.g. something like this, but with better formatting
```suggestion
					if ($item['name'] === $breakpoint 
					&& $item['type'] === $type 
					&& (!empty((string) $attributeValue) 
						|| gettype($attributeValue) === 'integer' 
						|| gettype($attributeValue) === 'float' 
						|| gettype($attributeValue) === 'double' 
						|| $attributeValue === '0')
					) { 
```